### DISCUSSION TITLE
Atom Unihertz continuous disconnections

### DISCUSSION THREAD

**Discussion by azizmarafie** [NightscoutFoundation/xDrip#2991] [Bug reports]
Hello
i am posting in here after Navid Fo has suggested to do so

I am using G6 CGM and Xdrip +
For the past 2 weeks I do facing a continuous discounts for 30-49 mins each time

I have changed the transmitter several times

I am using unihertz Mobile atom
Running on android version 9

The mobile build number is
Unihertz_atom_eea_20200303

When I try the transmitter on two other mobiles that has dexcom app installed on and another mobile S8 that has xdrip+ both work fine

i have reviewed the setup with Navid Fo and all is ok.

any suggestion how can i solve this, i am using Atom mobile coz of the size which is suitable for my 8 years old kid
6 moths ago all was working fine until the one of the xdrip updates

![IMG_8893](https://github.com/NightscoutFoundation/xDrip/assets/26650378/0ed62886-b0ce-4b4e-9912-dac6cb957a11)
![IMG_8895](https://github.com/NightscoutFoundation/xDrip/assets/26650378/5a4f2359-55ba-427f-9a2f-c0d2cf0dd501)
![IMG_8894](https://github.com/NightscoutFoundation/xDrip/assets/26650378/b970ef79-43bc-4667-9d33-9de311436e40)
![IMG_8896](https://github.com/NightscoutFoundation/xDrip/assets/26650378/e6f23b68-ed55-4ce6-88b7-04b5edd822a2)

**Comment by Navid200** [comment]
You updated xDrip to the latest I thought.
But, your classic status page shows a very old release.

This tells you how to disable the power saving setting on that phone:
https://github.com/NightscoutFoundation/xDrip/issues/734#issuecomment-463128661

Unfortunately, according to this (https://github.com/NightscoutFoundation/xDrip/issues/734#issuecomment-464678089), it looks like this is not a very good phone because of its broken Bluetooth stack.
  
If I were you, I would update xDrip to the current Nightly and would not go back to the very old release you are currently using according to your screenshots.
Then, I would disable power saving.
Then, please tell us what happens.

**Reply by azizmarafie** [reply]
hello

the images which are attached are old image i have updated the xdrip to the 13.07.2023 version when i was discussing with you on facebook see attached image

the phone used to work fine for 3 years until the past 6 months
i am not sure if it is from the latest versions of the transmitter or the xdrip update ?

![IMG_8925](https://github.com/NightscoutFoundation/xDrip/assets/26650378/697d5361-d481-464b-90ad-dfbfb36da7e0)

**Reply by Navid200** [reply]
There are other developers here and the reason I asked you to post here is to take advantage of their knowledge.
They are not aware of our previous conversations.  Therefore, it is not helpful to the objective if you post an outdated image.

Anyway, I believe the problem is the phone you are using.
Hopefully, someone can point out another small phone that does not have a messed up Bluetooth stack.

**Reply by Navid200** [reply]
You can install an older version of xDrip and try if you think it's xDrip that has changed.

**Reply by Navid200** [reply]
Did you disable power saving as explained here? 

Go in to `Android Settings` -> `Smart Assist` -> `Power Save Manager` -> Tick xDrip on in the whitelist

**Reply by azizmarafie** [reply]
The power saver on Xdrip is disabled long time ago 
![IMG_8939](https://github.com/NightscoutFoundation/xDrip/assets/26650378/8f4e93c7-b475-4f99-9170-8de208c48ca1)
![IMG_8938](https://github.com/NightscoutFoundation/xDrip/assets/26650378/061cb677-a8d7-479e-b6d6-f5f625bbfaa2)
![IMG_8937](https://github.com/NightscoutFoundation/xDrip/assets/26650378/5e5129eb-a3ef-4f17-ae88-e0b7d6009492)
![IMG_8936](https://github.com/NightscoutFoundation/xDrip/assets/26650378/f160e589-e3db-4bfb-9f8b-e9a698f134dd)
![IMG_8935](https://github.com/NightscoutFoundation/xDrip/assets/26650378/d11b0f88-9e92-40af-9c57-ec44a9baf5a5)
![IMG_8934](https://github.com/NightscoutFoundation/xDrip/assets/26650378/32dd8026-4d76-45a1-9e5f-08b01091907a)
![IMG_8933](https://github.com/NightscoutFoundation/xDrip/assets/26650378/bbefd041-2532-4d88-bf77-608bc261a512)
![IMG_8932](https://github.com/NightscoutFoundation/xDrip/assets/26650378/a6fb886e-554a-4ad9-8742-40744a4d0c42)

**Reply by Navid200** [reply]
So, you don't see `Smart Assist` in Android settings?

**Reply by azizmarafie** [reply]
Yes I didn’t see I tried to search for it but never found it 

By the way i just downloaded the feb version and for the past 30 mins it’s working fine 

hoping it would continue working 

![IMG_8950](https://github.com/NightscoutFoundation/xDrip/assets/26650378/b06b29a1-add1-439c-8138-b57fe9ef52ba)

**Reply by azizmarafie** [reply]
@Navid200 the old version has continuous sensor error …. This is error related to Bluetooth 
![fcdf9f65-5103-43fb-b16b-19c54d5653c4](https://github.com/NightscoutFoundation/xDrip/assets/26650378/b09019a7-3c83-4095-81cd-e32b0a269136)

**Reply by Navid200** [reply]
So, using an older version of xDrip did not change anything and the problem remained, right?

**Reply by azizmarafie** [reply]
Hi navid, 
No, Changing to an older version of Xdrip solved the disconnecting issue. It is working fine now.

**Reply by azizmarafie** [reply]
The sensor error is not related to the disconnecting

**Reply by Navid200** [reply]
Well, we need to communicate better if we want to make progress.

This is what you typed:
the old version has continuous sensor error

Reading that, what I understand is that the old xDrip version has continuous sensor errors.

Why did you write that if you wanted to say that the old version has solved the problem?

**Reply by azizmarafie** [reply]
In the old version we used to get a lot of sensor error

in the new version we do not get the sensor error but we have lots of  disconnection problem, 

I am not sure if both are coming from the same error but each version reports the error differently. 

Mybe I assumed that both (sensor error & disconnections) are from the same problem 

from my experience the sensor error are due to the fast raise or fall of the BG.

Until now the old version is working fine.

**Reply by Navid200** [reply]
Understood now.  Thanks.

I'm looking at my release notes and I believe if you update to the following, it should still behave the same as the Feb. 20 version you have been using now.
https://github.com/NightscoutFoundation/xDrip/releases/tag/2023.04.10

Please let me know if you update to it, if it still continues to work or not.

**Reply by azizmarafie** [reply]
@Navid200 
It has been almost 20 hours now since I installed the version which you highlighted and it works fine 
And has no disconnection until now 

thanks for your assistance

**Reply by Navid200** [reply]
No, you are assisting us.  
Thanks

When you can, if it's OK, please now try this release:  https://github.com/NightscoutFoundation/xDrip/releases/tag/2023.07.01

Please let me know if this one also works or if this one breaks it for you.

**Reply by azizmarafie** [reply]
@Navid200 

i installed the version 2023.07.01 and it is working fine until now it has been 8 hours 

regards

**Reply by Navid200** [reply]
Let's monitor that for a full 24 hours please.
If it is till fine, please try this one:  https://github.com/NightscoutFoundation/xDrip/releases/tag/2023.07.02

**Reply by azizmarafie** [reply]
@Navid200 
Hello

2 hours ago I have installed version 2023.07.02 and it is working fine until now 

will notify you what happens within the next 10 hours

**Reply by Navid200** [reply]
If this version also works with no issues long enough, please switch to this:
https://github.com/NightscoutFoundation/xDrip/releases/tag/2023.07.06

**Reply by azizmarafie** [reply]
@Navid200 

just now switch to the 2023.07.06 will notify you in couple of hours 

regards
Talal

**Reply by azizmarafie** [reply]
@Navid200 

Hello

the version 2023.07.06 is working fine until now for the past 17 hours 

regards

**Reply by Navid200** [reply]
I need you to stay on that for a day.  Please make sure it is not breaking.
The unexpected thing, at least for me, is that since that version, there has been no change to the G6/Dex1 collector.

Anyway, let's wait a day and we will take it from there.
Thanks for your patience.

**Reply by Navid200** [reply]
Sorry, I meant a day or two, not a week

**Reply by azizmarafie** [reply]
@Navid200 

Ok, thanks will notify you if any changes happen 

regards

**Reply by azizmarafie** [reply]
@Navid200 

hello
all Is fine until now it has been almost 2 days with no disconnection 


Regards

**Reply by Navid200** [reply]
I am happy to know that.
Let's please move on to this version:  https://github.com/NightscoutFoundation/xDrip/releases/tag/2023.07.08

**Reply by azizmarafie** [reply]
@Navid200 
Hello
I have updated to 3023.07.08 7 hours ago 
And all is fine
Regards

**Reply by azizmarafie** [reply]
@Navid200 

Hello 

all is fine until now 

Any further instructions?

regards

**Reply by Navid200** [reply]
Please try this now:
https://github.com/NightscoutFoundation/xDrip/releases/tag/2023.07.13

**Reply by azizmarafie** [reply]
@Navid200 

I have this tried before 
And will try it again 

regards
Talal

**Reply by azizmarafie** [reply]
@Navid200 
Hello
Hi tested for the past 3 days 

Version 2023.07.13 works fine.

regards
Talal

**Reply by Navid200** [reply]
That's great.
The release after that is our current stable release.
Do you want to try that gain now?  
Here it is:  https://github.com/NightscoutFoundation/xDrip/releases/tag/2023.07.16

**Reply by azizmarafie** [reply]
@Navid200 
Hi

i installed it 9 hours ago 
And it’s working fine 

Will update you end of day 

regards

**Reply by azizmarafie** [reply]
@Navid200 

Hello

the version is working fine and no disconnects 

waiting for your further instructions 

regards

**Reply by Navid200** [reply]
That is the current stable release.
If you are not having the disconnects that you were experiencing previously, I can only speculate that they may have been a coincidence and not related to any issues with the xDrip release you were using at the time of the problem.
Please continue to use the stable release.

If some new feature comes up in the release notes (https://navid200.github.io/xDrip/docs/ReleaseNotes.html) that you like to have, give it a try.  If the disconnect issue comes back, go back to the stable release.

If you can recreate the problem by using a specific new release and resolve it consistently by going back to the stable release, please report it and we can have a look.

Thanks
